### PR TITLE
Implemented an iOS http plugin and the unit testing iOS project.

### PR DESCRIPTION
**Pull request by v-hoccha**
Added three projects for Json, XPlatCppSdk, Test iOS application.
Implemented iOS http plugin.

### CODE REVIEW

File: build/iOS/TestIOSApp/UI/ViewController.m [hunk 1]
```diff
@@ -0,0 +1,201 @@
+//
+//  ViewController.m
+//  TestIOSApp
+//
+//  Created by Raymond Cha on 3/19/19.
```
Comment (alex-xbox) [inline build/iOS/TestIOSApp/UI/ViewController.m]: Please remove this line

**Comment by v-hoccha** [comment]
The changing test/TestApp/TestAppPch.cpp is due to the file encoding type. It was UTF16 encoding files and the Xcode can't accept it. So it is changed to Plain text encoding.

File: build/iOS/TestIOSApp/UI/ViewController.m [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        cell.backgroundColor = [UIColor colorWithRed:1.0 green:1.0 blue:1.0 alpha:1.0];
+    } else {
+        cell.backgroundColor = [UIColor colorWithRed:0.8 green:0.8 blue:0.8 alpha:1.0];
+    }
+    cell.textLabel.text = [self getLogStringWithIndex:indexPath.row];
+
+    return cell;
+}
+
+/*
```
Comment (alex-xbox) [inline build/iOS/TestIOSApp/UI/ViewController.m]: Why we have commented code here?

File: build/iOS/TestIOSApp/UI/main.m [hunk 3]
```diff
@@ -0,0 +1,16 @@
+//
+//  main.m
+//  TestIOSApp
+//
+//  Created by Raymond Cha on 3/19/19.
```
Comment (alex-xbox) [inline build/iOS/TestIOSApp/UI/main.m]: Please remove

File: build/iOS/TestIOSApp/log_bridge.c [hunk 4]
```diff
@@ -0,0 +1,36 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#include <stdio.h>
+#include <stdlib.h>
+
```
Comment (alex-xbox) [inline build/iOS/TestIOSApp/log_bridge.c]: reason for empty line?

File: build/iOS/TestIOSApp/UI/ViewController.m [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+    SetLogOutputCallback(DebugMessageToViewCallbackFunction, (__bridge void*)(self));
+    [self performSelector:@selector(processUnitTest:) withObject:[NSNumber numberWithInteger:0] afterDelay:1.0];
+}
+
+- (void)dealloc {
+    [NSObject cancelPreviousPerformRequestsWithTarget:self];
+    ResetLogOutputCallback();
+}
+
+- (NSLock *)getLogListLock {
```
Comment (alex-xbox) [inline build/iOS/TestIOSApp/UI/ViewController.m]: nit: code is inconsistent with bracket usage. Sometimes on new line, sometimes on the same line. Please use one style, according to language and code around you.

File: build/iOS/TestIOSApp/log_bridge.c [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+#include "log_bridge.h"
+
+static struct _LogOutputCallbackInfo {
+    LogOutputCallback callback;
+    LogOutputContext context;
+} s_callbackInfo = {};
+
+void
```
Comment (alex-xbox) [inline build/iOS/TestIOSApp/log_bridge.c]: Function declaration should look different, please change according to code style.

File: build/iOS/TestIOSApp/UI/ViewController.m [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+}
+*/
+
+/*
+// Override to support conditional rearranging of the table view.
+- (BOOL)tableView:(UITableView *)tableView canMoveRowAtIndexPath:(NSIndexPath *)indexPath {
+    // Return NO if you do not want the item to be re-orderable.
+    return YES;
+}
+*/
```
Comment (alex-xbox) [inline build/iOS/TestIOSApp/UI/ViewController.m]: Why comments left here?

File: code/source/playfab/OneDSIOSHttpPlugin.mm [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        OneDSCallRequestContainer& requestContainer = requestTask.OneDSRequestContainer();
+        payloadSize = static_cast<size_t>(requestContainer.requestBinaryBody.size());
+        payload = requestContainer.requestBinaryBody.data();
+        return true;
+    }
+
+    void OneDSIOSHttpPlugin::ProcessResponse(RequestTask& requestTask, const int httpCode)
+    {
+        OneDSCallRequestContainer& requestContainer = requestTask.OneDSRequestContainer();
+        if ((httpCode >= 200 && httpCode < 300) || httpCode == 0)
```
Comment (pgilmorepf) [inline code/source/playfab/OneDSIOSHttpPlugin.mm]: It's odd to me that an HTTP code of zero is "succeeded".

**Comment by pgilmorepf** [comment]
I have no comments, it looks good to start porting over to SdkGenerator, where it will actually be committed and landed.
Thanks!

**Pull request closed without merging**